src/control/server/storage_nvme.go

```diff
@@ -132,10 +135,12 @@ func (s *spdkSetup) reset() error {
 // for accessing Nvme devices (API) as well as storing device
 // details.
 type nvmeStorage struct {
-	env         spdk.ENV       // SPDK ENV interface
-	nvme        spdk.NVME      // SPDK NVMe interface
-	spdk        SpdkSetup      // SPDK shell configuration interface
-	config      *Configuration // server configuration structure
+	log         logging.Logger
+	ext         External
+	env         spdk.ENV  // SPDK ENV interface
+	nvme        spdk.NVME // SPDK NVMe interface
+	spdk        SpdkSetup // SPDK shell configuration interface
+	shmID       int
```

Comment:
this shouldn't be here as its IO server specific and should be passed in

Reply:
Each IOServerInstance has its own BdevProvider now. This may be reworked to having a BdevProvider instantiated at time of use, not sure. We'll see how the privileged binary work shakes out.